### PR TITLE
Feature/mx 1641 model update ifsg

### PR DESCRIPTION
**Pull request by erichesse**
# PR Context
- update ifsg to model v3 according to mex-assets/pull-requests/166
- make identifierInPrimarySource unique to avoid stableTargetId collisions according to mex-assets/pull-requests/182/
- metadata was sent for review
- fixed odk test, that needed matching with international projects

### CODE REVIEW

File: assets/mappings/__final__/ifsg/resource_disease.yaml [hunk 1]
```diff
@@ -83,81 +83,61 @@ isPartOf:
...
-  - "Lyme disease"
-  mappingRules:
-  - rule: "Assign 'en' as default for the language property of the text object."
-- fieldInPrimarySource: 'SpecimenName'
-  locationInPrimarySource: Meta.Disease
-  examplesInPrimarySource:
-  - "Bacillus anthracis"
-  - "Borelia burgdorferi"
-  mappingRules:
-  - rule: "Use value as it is. Use language detection."
```
Comment (esinsj) [inline assets/mappings/__final__/ifsg/resource_disease.yaml:138]: I think, these mappings (line 118-138) for "keyword" stayed in the resource_disease.yaml, they were just reformatted. And line 138 changed a bit.

File: assets/mappings/__final__/ifsg/resource_disease.yaml [hunk 2]
```diff
@@ -83,81 +83,61 @@ isPartOf:
...
   - setValues:
       - "https://mex.rki.de/item/language-1"
   comment: Deutsch
-publication:
-- fieldInPrimarySource: n/a
-  mappingRules:
-  - setValues:
-      - language: de
-        title: "Falldefinitionen"
-        url: "https://www.rki.de/DE/Content/Infekt/IfSG/Falldefinition/falldefinition_node.html"
```
Comment (esinsj) [inline assets/mappings/__final__/ifsg/resource_disease.yaml:151]: also the mapping for "publication" stays.

File: assets/mappings/__final__/ifsg/resource_disease.yaml [hunk 3]
```diff
@@ -83,81 +83,61 @@ isPartOf:
...
-  - rule: "Set a default by using the 'stable target id' of the organization-item with the name 'Robert Koch-Institut'."
+  - rule: "Set a default"
 resourceTypeGeneral:
 - fieldInPrimarySource: n/a
   mappingRules:
   - setValues:
-      - "https://mex.rki.de/item/resource-type-general-1"
+      - "https://mex.rki.de/item/resource-type-general-13"
   comment: Public Health Fachdaten
 resourceTypeSpecific:
```
Comment (esinsj) [inline assets/mappings/__final__/ifsg/resource_disease.yaml:142]: The "resourceTypeSpecific" mapping is dropped.

File: mex/extractors/ifsg/main.py [hunk 4]
```diff
@@ -207,13 +214,15 @@ def extracted_ifsg_resource_state(
     extracted_ifsg_resource_parent: ExtractedResource,
     extracted_primary_sources_ifsg: ExtractedPrimarySource,
     unit_stable_target_ids_by_synonym: dict[str, MergedOrganizationalUnitIdentifier],
+    meta_disease: list[MetaDisease],
```
Comment (esinsj) [inline mex/extractors/ifsg/main.py:217]: why do we need to extract the MetaDisease in the extracted_ifsg_resource_*state*? The descritption of this function is the same to the next one: extracted_ifsg_resource_*disease*.

File: mex/extractors/ifsg/transform.py [hunk 5]
```diff
@@ -97,6 +105,20 @@ def transform_resource_state_to_mex_resource(
         value["forValues"][0]: value["setValues"][0]
```
Comment (esinsj) [inline mex/extractors/ifsg/transform.py:105]: Hier wird durch  value["forValues"][0]: value["setValues"]**[0]** nur das erste dict (deutsche Sprache) ausgelesen, aber das zweite dict (englisch) ignoriert. "[0]" sollte daher gestrichen werden (?)

File: mex/extractors/ifsg/transform.py [hunk 6]
```diff
@@ -97,6 +105,20 @@ def transform_resource_state_to_mex_resource(
...
+    keyword = [
+        keyword
+        for keyword in [
+            *resource_state["keyword"][0]["mappingRules"][0]["setValues"],
+            *[row.disease_name for row in meta_disease if row.disease_name],
+            *[row.disease_name_en for row in meta_disease if row.disease_name_en],
+            *[row.specimen_name for row in meta_disease if row.specimen_name],
+        ]
+    ]
+
```
Comment (esinsj) [inline mex/extractors/ifsg/transform.py:121]: the keywords for disease_name and disease_name_en are lacking the language property.

File: mex/extractors/ifsg/transform.py [hunk 7]
```diff
@@ -107,25 +129,22 @@ def transform_resource_state_to_mex_resource(
         value["forValues"][0]: value["setValues"][0]
```
Comment (esinsj) [inline mex/extractors/ifsg/transform.py:129]: again a "[0]" too much

File: mex/extractors/ifsg/transform.py [hunk 8]
```diff
@@ -97,6 +105,20 @@ def transform_resource_state_to_mex_resource(
...
+            *resource_state["keyword"][0]["mappingRules"][0]["setValues"],
+            *[row.disease_name for row in meta_disease if row.disease_name],
+            *[row.disease_name_en for row in meta_disease if row.disease_name_en],
+            *[row.specimen_name for row in meta_disease if row.specimen_name],
+        ]
+    ]
+
     spatial_by_bundesland_id = None
     if resource_state["spatial"]:
         spatial_by_bundesland_id = {
```
Comment (esinsj) [inline mex/extractors/ifsg/transform.py:124]: for spatial there's also a "[0]" too much.

File: mex/extractors/ifsg/transform.py [hunk 9]
```diff
@@ -463,13 +511,14 @@ def transform_ifsg_data_to_mex_variables(
             item_row = item_by_id_item[id_item]
             value_set.append(item_row.item_name)
 
-        used_in = resource_disease_stable_target_id_by_id_type[row.id_type]
+        used_in = resource_disease_stable_target_id_by_id_type[str(row.id_type)]
         extracted_variables.append(
             ExtractedVariable(
                 belongsTo=belongs_to,
                 description=row.gui_tool_tip,
+                dataType=data_type_by_id[row.id_data_type],
```
Comment (esinsj) [inline mex/extractors/ifsg/transform.py:519]: Isn't the dataType of extractedVariables a DataTypeStr? But here it is an integer?

File: mex/extractors/ifsg/main.py (hunk 4 above)
Comment (erichesse) [reply mex/extractors/ifsg/main.py:217]: updated the docstrings:https://github.com/robert-koch-institut/mex-extractors/pull/236/commits/efca603af15a83544077bd01a566ec2f08c24a01

The keyword field of resource_state mapping states that MetaDisease rows are required

File: mex/extractors/ifsg/transform.py (hunk 5 above)
Comment (erichesse) [reply mex/extractors/ifsg/transform.py]: fixed https://github.com/robert-koch-institut/mex-extractors/pull/236/commits/266d4ac31a9bedfb629b7f177917fa6864fc4b8b

File: mex/extractors/ifsg/transform.py (hunk 6 above)
Comment (erichesse) [reply mex/extractors/ifsg/transform.py:130]: fixed https://github.com/robert-koch-institut/mex-extractors/pull/236/commits/266d4ac31a9bedfb629b7f177917fa6864fc4b8b

File: mex/extractors/ifsg/transform.py (hunk 7 above)
Comment (erichesse) [reply mex/extractors/ifsg/transform.py]: fixed https://github.com/robert-koch-institut/mex-extractors/pull/236/commits/266d4ac31a9bedfb629b7f177917fa6864fc4b8b

File: mex/extractors/ifsg/transform.py (hunk 8 above)
Comment (erichesse) [reply mex/extractors/ifsg/transform.py:133]: fixed https://github.com/robert-koch-institut/mex-extractors/pull/236/commits/266d4ac31a9bedfb629b7f177917fa6864fc4b8b

File: assets/mappings/__final__/ifsg/resource_disease.yaml (hunk 3 above)
Comment (erichesse) [reply assets/mappings/__final__/ifsg/resource_disease.yaml:142]: same here

**Pull request merged**